linode/rdns/framework_resource.go

```diff
@@ -0,0 +1,149 @@
+	"github.com/linode/terraform-provider-linode/linode/helper"
+)
+
+func NewResource() resource.Resource {
+	return &Resource{
+		BaseResource: helper.NewBaseResource(
+			helper.BaseResourceConfig{
+				Name:   "linode_rdns",
+				Schema: &frameworkResourceSchema,
+				IDType: types.StringType,
+			},
+		),
+	}
+}
+
+type Resource struct {
+	helper.BaseResource
+}
+
+func (r *Resource) Create(
+	ctx context.Context,
+	req resource.CreateRequest,
+	resp *resource.CreateResponse,
+) {
+	var data ResourceModel
+	client := r.Meta.Client
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	updateOpts := linodego.IPAddressUpdateOptions{
+		RDNS: data.RDNS.ValueStringPointer(),
+	}
+
+	ip, err := client.UpdateIPAddress(ctx, data.Address.ValueString(), updateOpts)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Failed to create Linode RDNS",
+			err.Error(),
+		)
+		return
+	}
+
+	data.parseIP(ip)
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *Resource) Read(
+	ctx context.Context,
+	req resource.ReadRequest,
+	resp *resource.ReadResponse,
+) {
+	client := r.Meta.Client
+
+	var data ResourceModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	ip, err := client.GetIPAddress(ctx, data.ID.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Failed to read the Linode RDNS", err.Error(),
+		)
+		return
+	}
+
+	data.parseIP(ip)
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *Resource) Update(
+	ctx context.Context,
+	req resource.UpdateRequest,
+	resp *resource.UpdateResponse,
+) {
+	var state, plan ResourceModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	client := r.Meta.Client
+
+	var updateOpts linodego.IPAddressUpdateOptions
+
+	resourceUpdated := false
+
+	if !state.RDNS.Equal(plan.RDNS) {
+		updateOpts.RDNS = plan.RDNS.ValueStringPointer()
+		resourceUpdated = true
+	}
+
+	if resourceUpdated {
+		ip, err := client.UpdateIPAddress(ctx, state.Address.ValueString(), updateOpts)
+		if err != nil {
+			resp.Diagnostics.AddError(
+				"Failed to update the Linode RDNS",
+				err.Error(),
+			)
+			return
+		}
+
+		state.parseIP(ip)
+		resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+	}
+}
+
+func (r *Resource) Delete(
+	ctx context.Context,
+	req resource.DeleteRequest,
+	resp *resource.DeleteResponse,
+) {
+	var data ResourceModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	client := r.Meta.Client
+
+	updateOpts := linodego.IPAddressUpdateOptions{
+		RDNS: nil,
+	}
+
+	_, err := client.UpdateIPAddress(ctx, data.Address.ValueString(), updateOpts)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Failed to delete Linode RDNS",
+			err.Error(),
+		)
```

Comment:
Should ignore 404? When the user delete linode first and then the rdns, even if the rdns was gone with the linode, the deletion can still be considered as success, right?

Reply:
Great catch!